hail/python/test/hail/linalg/test_linalg.py

```diff
@@ -169,6 +169,14 @@ def test_from_entry_expr_simple(self):
             a4 = hl.eval(BlockMatrix.read(path).to_ndarray())
             self._assert_eq(a1, a4)
 
+    def test_from_entry_expr_empty_parts(self):
+        with hl.TemporaryDirectory(ensure_exists=False) as path:
+            mt = hl.balding_nichols_model(n_populations=5, n_variants=2000, n_samples=20, n_partitions=200)
+            mt = mt.filter_rows((mt.locus.position <= 500) | (mt.locus.position > 1500)).checkpoint(path)
```

Comment:
unfortunately, this is a bit of a load bearing checkpoint, the issue we've run into doesn't replicate otherwise